### PR TITLE
ARC-677 permissions doc update

### PR DESCRIPTION
**Pull request by rachellerathbone**
Updated permissions section of the README and added an FAQs doc so we can start referencing common customer concerns. Please review both docs to check for typos and make sure links look like they point to the right place. You won't be able to access my question to the GH team about write permissions on deployment creation events because 'Akismet' has apparently made it temporarily private. :shakes-fist:

Screen Shot 2021-12-01 at 1 09 15 pm

### CODE REVIEW

File: README.md
Comment (thombergs) [inline README.md:116]:
```suggestion
|**Read** access to contents & metadata |**Contents** (aka code) and **metadata** are needed to sync development information to Jira. <br><br> **Medadata:** All GitHub apps have read-only metadata permission set by default. This is a [mandatory requirement by GitHub](https://docs.github.com/en/rest/reference/permissions-required-for-github-apps#metadata-permissions) and is needed to provide access to a collection of read-only endpoints with metadata for various resources. These endpoints do not provide sensitive private repository information. Read-only metadata permissions are used for the following webhook: <br> - repository (all events excluding `deleted`) <br><br> **Contents:** Read-only content permissions are used for the following webhooks: <br> - commit comment <br> - create <br> - delete <br> - push <br> - workflow run|
```

File: README.md
Comment (thombergs) [inline README.md:117]:
```suggestion
|**Read** and **write** access to issues and pull requests|**Issues** and **pull requests** are used by the GitHub for Jira app to power Smart Commit actions and unfurl Jira URLs. "Unfurling" means that the app looks for Jira issue keys like `[ABC-123]` in pull request or issue comments and then replaces those issue keys with a link to the respective Jira issue. <br><br> **Issues:** Read and write issue permissions are used for the following webhooks: <br> - issue comment <br> - issues <br><br> **Pull requests:** Read and write pull request permissions are used for the following webhooks: <br> - pull request <br> - pull request review|
```

File: README.md
Comment (thombergs) [inline README.md:118]: The app is not calling the API to create (POST) deployment statuses (it shouldn't be, at least!). It should be calling [this API](https://docs.github.com/en/rest/reference/repos#list-deployment-statuses) to list deployment status. The docs say that this only requires "pull access". Is this "pull access" the missing piece perhaps? Can we give "pull" access without giving "write" access to deployments?

File: docs/FAQs.md
Comment (thombergs) [inline docs/FAQs.md:5]: Maybe make each question an `<h3>` to make them more prominent (and ditch the "Permissions" header, because we only seem to have one category anyways. That way we wouldn't need to use ugly `<br>`s in Markdown :)

Comment (thombergs) [inline docs/FAQs.md:5]:
```suggestion
**Q:** The permission scope for code and metadata suggests data is read to synchronize development information. Is code stored on Jira? Should I be concerned that, were the app to be compromised, an actor can exfiltrate all our code from GitHub?
```

File: docs/FAQs.md
Comment (thombergs) [inline docs/FAQs.md:7]: Maybe make it more clear in the answer that the app does not store code. The answer explains what is happening, but it's not really answering the question "is code stored in Jira?"

File: docs/FAQs.md
Comment (thombergs) [inline docs/FAQs.md:29]: The docs say that an app needs write access if you want to call the `POST` endpoint to CREATE a `deployment_status` event. But the app isn't doing that.

File: docs/FAQs.md
Comment (thombergs) [inline docs/FAQs.md:35]: "There are no consequences as such" is a lie :). The consequence is that you won't see deployments in Jira

**Review by thombergs — COMMENTED** [review]
Nice! I still don't understand why the app would need write permissions for deployments, though (see my comments).

File: README.md
Comment (rachellerathbone) [reply README.md:118]: No, it's not making a POST request but `deployment_status` is fired when a deployment is created https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#deployment_status and, according to the GitHub documentation, GitHub apps require read and write access https://docs.github.com/en/rest/reference/repos#create-a-deployment-status

File: docs/FAQs.md
Comment (rachellerathbone) [reply docs/FAQs.md:5]: Reading this again, 'could' would make more grammatical sense, would it not?

File: docs/FAQs.md
Comment (rachellerathbone) [reply docs/FAQs.md:29]: From what I've read, [GitHub apps require read & write](https://docs.github.com/en/rest/reference/repos#create-a-deployment-status) to access [deployment_status](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#deployment_status) which is considered a creation event.

File: README.md
Comment (thombergs) [reply README.md:118]: The deployment is created by a GitHub action or some external tool like Jenkins. _That_ app needs write access to create a deployment, which causes the `deployment_status` webhook, which gets sent to the GitHub app. 

I understand the docs that write access is required for _creating a deployment_, which makes sense. But the GitHub app is not creating a deployment. It's consuming `deployment_status` events.

File: docs/FAQs.md
Comment (thombergs) [reply docs/FAQs.md:23]: see my comment above, I don't understand the docs like that :)

File: README.md
Comment (thombergs) [reply README.md:118]: Looking into the docs of the `deployment_status` event, I see this:

image

Sadly, this only says `deployments` permission on not whether it's read or write.

File: docs/FAQs.md
Comment (KAbakumovAtl) [reply docs/FAQs.md:23]: Do we use that endpoint? As Tom saying it seems that is being used for creating deployments within GitHub, while we only need to know which deployments are there. 

However I also remember @atrigueiro was saying that it didn't work without write permissions, so we had to add it. Seems we need it, but I am not sure that the mentioned above endpoint is the reason why.

**Review by mboudreau — APPROVED** [review]

File: README.md
Comment (yvonnevictoria) [inline README.md:140]: Update this URL before merging to point to `master`

**Review by yvonnevictoria — APPROVED** [review]

**Pull request merged**
